### DISCUSSION TITLE
集群部署时sf.init()超时

### DISCUSSION THREAD

**Discussion by jianjew** [secretflow/secretflow#1307] [Q&A]
设置8085为ray端口， 8086为address端口， 8087为listen_add端口, 都是未使用的端口
通过docker部署secretflow时，ray集群已经搭建成功了，调用sf.init() 两个容器中都报如下错误（使用代码贴在后面）：
2024-05-28 08:31:12.926 INFO barriers.py:520 [bob] -- [Anonymous_job] Try ping ['alice'] at 193 attemp, up to 3600 attemps.
2024-05-28 08:31:15.929 INFO barriers.py:520 [bob] -- [Anonymous_job] Try ping ['alice'] at 194 attemp, up to 3600 attemps.
(SenderProxyActor pid=276) 2024-05-28 08:31:15.852 ERROR barriers.py:171 [bob] -- [Anonymous_job] Failed to send data to seq_id ping of alice from ping, error: <AioRpcError of RPC that terminated with:
(SenderProxyActor pid=276)      status = StatusCode.DEADLINE_EXCEEDED
(SenderProxyActor pid=276)      details = "Deadline Exceeded"
(SenderProxyActor pid=276)      debug_error_string = "UNKNOWN:Deadline Exceeded {grpc_status:4, created_time:"2024-05-28T08:31:15.851275266+00:00"}"
(SenderProxyActor pid=276) >
(SenderProxyActor pid=276) 2024-05-28 08:31:18.855 ERROR barriers.py:171 [bob] -- [Anonymous_job] Failed to send data to seq_id ping of alice from ping, error: <AioRpcError of RPC that terminated with:
(SenderProxyActor pid=276)      status = StatusCode.DEADLINE_EXCEEDED
(SenderProxyActor pid=276)      details = "Deadline Exceeded"
(SenderProxyActor pid=276)      debug_error_string = "UNKNOWN:Deadline Exceeded {created_time:"2024-05-28T08:31:18.855214362+00:00", grpc_status:4}"
(SenderProxyActor pid=276) >
2024-05-28 08:31:18.935 INFO barriers.py:520 [bob] -- [Anonymous_job] Try ping ['alice'] at 195 attemp, up to 3600 attemps.
(SenderProxyActor pid=276) 2024-05-28 08:31:21.855 ERROR barriers.py:171 [bob] -- [Anonymous_job] Failed to send data to seq_id ping of alice from ping, error: <AioRpcError of RPC that terminated with:
(SenderProxyActor pid=276)      status = StatusCode.DEADLINE_EXCEEDED
(SenderProxyActor pid=276)      details = "Deadline Exceeded"
(SenderProxyActor pid=276)      debug_error_string = "UNKNOWN:Deadline Exceeded {grpc_status:4, created_time:"2024-05-28T08:31:21.855192639+00:00"}"
(SenderProxyActor pid=276) >
2024-05-28 08:31:21.937 INFO barriers.py:520 [bob] -- [Anonymous_job] Try ping ['alice'] at 196 attemp, up to 3600 attemps.
2024-05-28 08:31:24.940 INFO barriers.py:520 [bob] -- [Anonymous_job] Try ping ['alice'] at 197 attemp, up to 3600 attemps.
(SenderProxyActor pid=276) 2024-05-28 08:31:24.864 ERROR barriers.py:171 [bob] -- [Anonymous_job] Failed to send data to seq_id ping of alice from ping, error: <AioRpcError of RPC that terminated with:
(SenderProxyActor pid=276)      status = StatusCode.DEADLINE_EXCEEDED
(SenderProxyActor pid=276)      details = "Deadline Exceeded"
(SenderProxyActor pid=276)      debug_error_string = "UNKNOWN:Deadline Exceeded {grpc_status:4, created_time:"2024-05-28T08:31:24.864255521+00:00"}"
(SenderProxyActor pid=276) >
(SenderProxyActor pid=276) 2024-05-28 08:31:27.866 ERROR barriers.py:171 [bob] -- [Anonymous_job] Failed to send data to seq_id ping of alice from ping, error: <AioRpcError of RPC that terminated with:
(SenderProxyActor pid=276)      status = StatusCode.DEADLINE_EXCEEDED
(SenderProxyActor pid=276)      details = "Deadline Exceeded"
(SenderProxyActor pid=276)      debug_error_string = "UNKNOWN:Deadline Exceeded {created_time:"2024-05-28T08:31:27.866268081+00:00", grpc_status:4}"
(SenderProxyActor pid=276) >
2024-05-28 08:31:27.943 INFO barriers.py:520 [bob] -- [Anonymous_job] Try ping ['alice'] at 198 attemp, up to 3600 attemps.
(SenderProxyActor pid=276) 2024-05-28 08:31:30.867 ERROR barriers.py:171 [bob] -- [Anonymous_job] Failed to send data to seq_id ping of alice from ping, error: <AioRpcError of RPC that terminated with:
(SenderProxyActor pid=276)      status = StatusCode.DEADLINE_EXCEEDED
(SenderProxyActor pid=276)      details = "Deadline Exceeded"
(SenderProxyActor pid=276)      debug_error_string = "UNKNOWN:Deadline Exceeded {grpc_status:4, created_time:"2024-05-28T08:31:30.866275255+00:00"}"
(SenderProxyActor pid=276) >
2024-05-28 08:31:30.949 INFO barriers.py:520 [bob] -- [Anonymous_job] Try ping ['alice'] at 199 attemp, up to 3600 attemps.
(SenderProxyActor pid=276) 2024-05-28 08:31:33.870 ERROR barriers.py:171 [bob] -- [Anonymous_job] Failed to send data to seq_id ping of alice from ping, error: <AioRpcError of RPC that terminated with:
(SenderProxyActor pid=276)      status = StatusCode.DEADLINE_EXCEEDED
(SenderProxyActor pid=276)      details = "Deadline Exceeded"
(SenderProxyActor pid=276)      debug_error_string = "UNKNOWN:Deadline Exceeded {created_time:"2024-05-28T08:31:33.870269282+00:00", grpc_status:4}"
(SenderProxyActor pid=276) >
2024-05-28 08:31:33.953 INFO barriers.py:520 [bob] -- [Anonymous_job] Try ping ['alice'] at 200 attemp, up to 3600 attemps.

代码：
1、环境准备

    # 安装docker
    # 两台虚拟机：
    192.168.1.128（alice）
    192.168.1.100（bob）
    # 采用docker方式部署
    
    # 对应的通信地址设置如下：
    如图所示，存在三套地址：
    Ray头节点的监听地址
        alice 192.168.1.128:8085
        bob 192.168.1.100:8085
    Proxy的通信地址
        alice 192.168.1.128:8086
        bob 192.168.1.100:8086
    SPU的通信端口
        alice 8087
        bob 8087

2、安装secretflow

    # 在两台机器上分别执行：
    docker pull secretflow/secretflow-lite-anolis8:latest
    docker run -itd --network=host --name=alice secretflow/secretflow-lite-anolis8 

3、构建ray环境

    # 使用docker exec命令进入容器，启动ray,使用8085端口
    # netstat -anp | grep 8085 保证8085端口没有被占用
    # alice:
    ray start --head \
    --node-ip-address="192.168.1.128" \
    --port="8085" \
    --resources='{"alice": 16}' \
    --include-dashboard=False \
    --disable-usage-stats
    
    # bob
    ray start --head \
    --node-ip-address="192.168.1.100" \
    --port="8085" \
    --resources='{"bob": 16}' \
    --include-dashboard=False \
    --disable-usage-stats
    
    # 注： 当屏幕中输出 “Ray runtime started.” 意味着主节点成功启动

4、集群配置初始化

两方分别执行如下代码：

    # alice run
    import secretflow as sf
    cluster_config ={
        'parties': {
            'alice': {
                # replace with alice's real address.
                'address': '192.168.1.128:8086',  # proxy通信地址
                'listen_addr': '0.0.0.0:8087'  # SPU通信地址
            },
            'bob': {
                # replace with bob's real address.
                'address': '192.168.1.100:8086',
                'listen_addr': '0.0.0.0:8087'
            },
        },
        'self_party': 'alice'
    }
    
    sf.shutdown()  # 销毁现有的集群
    # Use ray head adress please. address='Ray head node address'
    sf.init(address='192.168.1.128:8085', cluster_config=cluster_config)   # Ray通信地址
    

    # bob run
    import secretflow as sf
    cluster_config ={
        'parties': {
            'alice': {
                # replace with alice's real address.
                'address': '192.168.1.128:8086',  # proxy通信地址
                'listen_addr': '0.0.0.0:8087'  # SPU通信地址
            },
            'bob': {
                # replace with bob's real address.
                'address': '192.168.1.100:8086',
                'listen_addr': '0.0.0.0:8087'
            },
        },
        'self_party': 'bob'
    }
    
    sf.shutdown()  # 销毁现有的集群
    # Use ray head adress please. address='Ray head node address'
    sf.init(address='192.168.1.100:8085', cluster_config=cluster_config)   # Ray通信地址

**Comment by Chrisdehe** [comment]
已在https://github.com/secretflow/secretflow/issues/1308
解决
